db/system_keyspace.cc

```diff
@@ -614,8 +614,15 @@ schema_ptr system_keyspace::built_indexes() {
             {"bytes_out", long_type},
             {"columnfamily_name", utf8_type},
             {"compacted_at", timestamp_type},
+            {"compaction_type", utf8_type},
             {"keyspace_name", utf8_type},
             {"rows_merged", map_type_impl::get_instance(int32_type, long_type, true)},
+            {"shard_id", int32_type},
+            {"sstables_in", list_type_impl::get_instance(sstableinfo_type, false)},
+            {"sstables_out", list_type_impl::get_instance(sstableinfo_type, false)},
+            {"total_tombstone_purge_attempt", long_type},
+            {"total_tombstone_purge_failure_due_to_overlapping_with_memtable", long_type},
+            {"total_tombstone_purge_failure_due_to_overlapping_with_uncompacting_sstable", long_type},
```

Comment:
These aren't conditional.

(we should have a helper that allows building schemas where some columns are tied to a feature&)

Reply:
Why should they be? 

I think that so far we have been adding columns unconditionally. The feature flag is used to ensure no data is added to the new columns until all nodes are upgraded. As long as nothing is written there, a rollback is possible.